First version

elisabettai opened the pull request.

elisabettai committed:
First version working on local deployment

elisabettai committed:
Update README.md

elisabettai committed:
Quiet install

elisabettai committed:
Merge branch 'first_version' of github.com:ITISFoundation/FSL into first_version

elisabettai committed:
Maximize disk space in github actions

elisabettai committed:
CI: add command to analyze disk usage

elisabettai committed:
Small cleanup in Dockerfile and in CI file

pcrespov commented on .github/workflows/check-image.yml:
```diff
@@ -0,0 +1,31 @@
+        uses: docker://itisfoundation/ci-service-integration-library:v1.0.1-dev-32
+        with:
+          args: ooil --version
+      - name: Assemble docker-compose spec
```
assemble? :-) did you write this ... sounds like ANE to me

GitHK replied:
It's starting to catch on. Be ready to assemble more stuff in the future!

elisabettai replied:
> what are you using the validation folders for ...?

I think that the validation folders are used for testing locally (I kept the structure from the UI service cookiecutter). 
Now I've added a file in the validation/inputs that can be used to test the application locally.

pcrespov reviewed (commented):
what are you using the validation folders for ...?

GitHK commented on README.md:
```diff
@@ -1,6 +1,6 @@
 # fsl
 
-FSL is a comprehensive library of analysis tools for FMRI, MRI and DTI brain imaging data, created by the Analysis Group, FMRIB, Oxford, UK.
+FSL is a comprehensive library of analysis tools for FMRI, MRI and DTI brain imaging data, created by the Analysis Group, FMRIB, Oxford, UK. See https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FSL for more information.
```
Regarding versioning below. there is no integration version to change. You might want to remove that.

GitHK replied:
also there is no `make version-service-*` only `make version-*` is available.

GitHK replied:
Also this is not gitlab, this is GitHub. I would remove the reference to that.

GitHK replied:
Also remove the gitlab reference, it is of no value to external users. I would drop it.

elisabettai replied:
Thanks, I've updated the README accordingly. Also added some instructions on how to test the application (mostly for us or whoever wants to deploy a new version).

elisabettai committed:
Updated README

elisabettai committed:
- Added same validation input data
- Validation folder is now mounted when running shell-app
- Solved libtinfo.so.6 version problem by changing LD_LIBRARY_PATH

mrnicegyu11 commented on app/Dockerfile:
```diff
@@ -0,0 +1,79 @@
+FROM ubuntu:20.04 AS base
```
minor: you could consider using the latest LTS ubuntu, this one is already an older version.

elisabettai replied:
Ok. I was discussing with Pedro about having a more standardized set of base images, so the building might be a bit faster. 
I'd be happy to adopt whatever we decide (apart from special cases, where using a more specialized image might be an advantage).

mrnicegyu11 commented on app/Dockerfile:
```diff
@@ -0,0 +1,79 @@
+ENV DEBIAN_FRONTEND=noninteractive
+
+
+RUN apt-get update -y && \
```
minor: one could consider pinning versions in the future, although I myself often dont do this either /o\

mrnicegyu11 commented on app/config/supervisord.conf:
```diff
@@ -0,0 +1,42 @@
+command=fsl
+autorestart=true
+stdout_logfile=/dev/fd/1
+stdout_logfile_maxbytes=0
```
minor: Maybe set this to a very high number instead of 0 which probably means unlimited

elisabettai replied:
Done (I'll put max vals of 50MB and 150 MB. I might update that in the cookiecutter for UI services as well.

mrnicegyu11 commented on app/config/supervisord.conf:
```diff
@@ -0,0 +1,42 @@
+environment=DISPLAY=:0
+autorestart=true
+stdout_logfile=/dev/fd/1
+stdout_logfile_maxbytes=0
```
vide infra

mrnicegyu11 commented on app/config/supervisord.conf:
```diff
@@ -0,0 +1,42 @@
+command=/usr/bin/Xtigervnc -desktop "FSL" -rfbport 5900 -SecurityTypes None -AlwaysShared -AcceptKeyEvents -AcceptPointerEvents -AcceptSetDesktopSize -SendCutText -AcceptCutText :0
+autorestart=true
+stdout_logfile=/dev/fd/1
+stdout_logfile_maxbytes=0
```
vide infra

mrnicegyu11 commented on web/config/nginx.conf:
```diff
@@ -0,0 +1,39 @@
+        proxy_set_header   Host $host;
+    }
+
+    # Hack until we have the dynamic sidecar
```
Dont we have the dynamic sidecar? :D Can we maybe get rid of this hack then?

mrnicegyu11 replied:
@GitHK @pcrespov  any more info on this?

GitHK replied:
@mguidon might be able to answer to you here. I'm not sure why these are being propagated

mrnicegyu11 reviewed (approved):
very good stuff Eli, awesome. Cheers!
